### PR TITLE
Add information about the control file to Concepts.md

### PR DESCRIPTION
**Pull request by MichalMal99**
New information with descriptions of the fields from the d/control file has been added to the Concepts.md. That information is compressed and helps to gain basic knowledge of the debian/control.

### CODE REVIEW

**Review by utkarsh2102 — CHANGES_REQUESTED** [review]
Why not just point to https://www.debian.org/doc/manuals/maint-guide/dreq.en.html#control? That's supposed to be the "official" docs. :)

**Comment by MichalMal99** [comment]
Hello Utkarsh! Thank you for your comment. There is an official document, but I want to outline everything in the handbook without going into too much detail so that users can find more information elsewhere (I can attach that link at the end). The goal of that handbook is to assist newcomers in adjusting to the team as well as the overall workflow and process, which is why audience is the first thing we consider here. I suppose that if this handbook were only for experienced team members, we would not create it at all, because they know what to look for and where to find specific information from other resources, but we collect here our points of view and write them down for people who joined (and obviously not only) and do not know where to find specific information or what is more important from the bunch of information. Each one receives the knowledge in a different way, which is why it's important to let everyone explain how they understand a specific topic. (For example, at the end, I've included a "Things to keep in mind..." section based on my previous experience and thought it was worth noting for the future. Moreover, outlining the most important information for first-time readers is not so overwhelming, and when they assimilate the basics, it's easier to go deeper into a particular topic.
To sum up, I would leave everything as-is and moreover add the link to the official doc at the end for further reading. What do you think about it?

File: Concepts.md [hunk 1]
```diff
@@ -288,6 +288,96 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+directory.
+Control files are used for source packages, binary packages and the
+.changes files that control the installation of uploaded files.
+Each control file contains a number of fields. Each field begins with a
+tag such as Package or Version.
+The tags are quite descriptive and to sum them up after checking the
+control file, we have a wide overview of the package.
+Each field gives us a new bit of knowledge about the package, such as
+package name, package type, version or description.
+Some of the fields are less descriptive and we can cover a bit of them.
```
[the remark this answers is not in the document]
Comment (cpaelzer) [reply Concepts.md:303]: Actually we are not even talking about the "most important" more like, "here just a few examples"

File: Concepts.md [hunk 2]
```diff
@@ -288,6 +288,96 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+unusual installations.
+
+9. Breaks
+
+Lists packages that this one breaks. So it helps us to make sure that
+everything within the package will work.
+
+10. Conflicts
+
+Lists packages that conflict with this one, for example by containing
```
Comment (cpaelzer) [inline Concepts.md:363]: Just pick up to three field as example, describe them shortly - the link to the full debian definition will do the rest.
It only is in the UMH to make everyone aware they do exist and where to find info about them.

File: Concepts.md [hunk 3]
```diff
@@ -288,6 +288,96 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+files with the same names.
+
+
+
+Thing to keep in mind while proposing something to the package:
```
Comment (cpaelzer) [inline Concepts.md:368]: while -> when

File: Concepts.md [hunk 4]
```diff
@@ -288,6 +288,96 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+
+There are also VCS entries which help us identify where the package is
+developed. There is some advice to consider if we want to propose
+something.
+E.g., we want to propose autopkgtests which are not present in Debian
+and package is developed in Git, but there is a maintainer from Debian.
+So in that case, we should send it to that specific person from debian,
+because autopkgtests exist only in Ubuntu and Debian and proposing them
+to the git repo is pointless.
```
Comment (cpaelzer) [inline Concepts.md:377]: not yet greta, we will re-eval once the rest is cleaned up

**Review by cpaelzer — CHANGES_REQUESTED** [review]
thanks, not yet where I wanted it. Ping me again once improved

**Comment by MichalMal99** [comment]
Done

File: Concepts.md [hunk 5]
```diff
@@ -288,6 +288,77 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+Each field gives us a new bit of knowledge about the package, such as
+package name, package type, version or description.
+You will find it all in way more detail by visiting: https://www.debian.org/doc/manuals/maint-guide/dreq.en.html#control
+Here are just a few examples of the most common tags:
+
+1. Maintainer
+
+This field typically gives us information on who created the package.
+If we have suggestions or want to propose something, we should talk
+to maintainer whose email is given in that field.
```
Comment (cpaelzer) [inline Concepts.md:306]: "the maintainer"

Comment (cpaelzer) [reply Concepts.md:306]: You might want to drop "Source" and "Depends" but add "VCS" instead.
The reason is that we wanted this to help "who/where to contact" for
- maintainer we can say to talk to them (you have that)
- for VCS we can state that there you find code and might propose a PR if on salsa
- for homepage (you mention it) we can say we find more contact or subject matter experts there

So these three would be great to have, the rest is covered by the link.

Maybe
"Here are just a few examples of the most common tags:"
to
"Here are just a few examples that help you who to contact if in doubt (which is a typical beginners question):"

File: Concepts.md [hunk 6]
```diff
@@ -288,6 +288,77 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+The name of the source package that this binary package came from, if it
+is different than the name of the package itself.
+
+4. Depends
+
+List of packages that are required for this package to provide
+non-trivial amount of functionality.
+
+
+Thing to keep in mind when proposing something to the package:
```
Comment (cpaelzer) [inline Concepts.md:323]: Thing ... Drop this sentence, it is odd here

File: Concepts.md [hunk 7]
```diff
@@ -288,6 +288,77 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+
+Source: ipmitool
```
Comment (cpaelzer) [inline Concepts.md:325]: Move the example up - it is good, but it will help to be before the file explanations

File: Concepts.md [hunk 8]
```diff
@@ -288,6 +288,77 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+ libfreeipmi-dev [!hurd-i386],
+ libreadline-dev,
+ libssl-dev
+Standards-Version: 4.6.1.0
+Rules-Requires-Root: no
+Vcs-Git: git://jff.email/opt/git/ipmitool.git
+Vcs-Browser: https://jff.email/cgit/ipmitool.git
+Homepage: https://github.com/ipmitool/ipmitool
+
+What we see above is part of the debian/control file of the ipmitool
```
Comment (cpaelzer) [inline Concepts.md:342]: Duplicate  - you explain it above at the field (which is good) - do not explain it again here.
Most of what you have here is great - carry what you really like up to the per field explanations.

File: Concepts.md [hunk 9]
```diff
@@ -288,6 +288,77 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+package.
+We can easily spot the entry, which tells us that the package is
+developed in git. We also see that the maintainer is the person from
+debian.
+There is an advice to consider.
+E.g., we want to propose autopkgtests which are not present in Debian,
```
Comment (cpaelzer) [inline Concepts.md:348]: Use full "For example .." and I like to have a "apply this" section at the end - this is good.

File: Concepts.md [hunk 10]
```diff
@@ -288,6 +288,77 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+but they are present in Ubuntu and specific package (in this case
+ipmitool) is developed in Git, but there is also the maintainer from
```
Comment (cpaelzer) [inline Concepts.md:350]: (in this case
￼ipmitool) - that is clear it is the example

Comment (cpaelzer) [reply Concepts.md:350]: why ", but there is also the maintainer from" there always is.
But implies something and it seems missing here.

Comment (cpaelzer) [reply Concepts.md:350]: That is you secenario:
For example - we want to propose autopkgtests which are not present in Debian, but they are present in Ubuntu.
Does this go to upstream, or Debian git or the maintainer?

File: Concepts.md [hunk 11]
```diff
@@ -288,6 +288,77 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+Debian.
+What we should do if there is a change which might be necessary only for
+Debian?
+So in that case, we should send it to that specific person from Debian
+(Jörg Frings-Fürst), not submit it to upstream, because autopkgtests
+exist only in Ubuntu and Debian, and proposing them to the git repository
+is pointless, because in that case, no one else apart from Debian
```
Comment (cpaelzer) [inline Concepts.md:357]: calrify "the git" - it is not salsa, it is upstream - if packaging is there it is actually the right git - but in this case it has no proposal function, that is why you fall back to mail.

Consider the order:
1. where? -> Debian or Upstream based on what you change
2. how? -> Debian bug / Salsa PR / email maintainer group / email maintainer (based on the situation)

**Review by cpaelzer — CHANGES_REQUESTED** [review]
round #2 - fight

File: Concepts.md (hunk 7 above)
Comment (cpaelzer) [reply Concepts.md:303]: done

File: Concepts.md [hunk 12]
```diff
@@ -288,6 +288,96 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
 We'll be creating our own changelog entry later and will discuss the various elements at that point.  But for now note how the first line contains the package name ('hello'), version number ('2.10-2ubuntu1'), and the Ubuntu release codename ('eoan').
 
 
+### Control file in debian directory
+
+Let's focus on the control file that exists inside the debian
+directory.
+Control files are used for source packages, binary packages and the
+.changes files that control the installation of uploaded files.
```
[the remark this answers is not in the document]
Comment (MichalMal99) [reply Concepts.md]: thanks, done

File: Concepts.md [hunk 13]
```diff
@@ -288,6 +288,96 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+Each control file contains a number of fields. Each field begins with a
```
[the remark this answers is not in the document]
Comment (MichalMal99) [reply Concepts.md]: thanks, done

File: Concepts.md [hunk 14]
```diff
@@ -288,6 +288,96 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+tag such as Package or Version.
+The tags are quite descriptive and to sum them up after checking the
```
[the remark this answers is not in the document]
Comment (MichalMal99) [reply Concepts.md]: thanks, done

File: Concepts.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichalMal99) [reply Concepts.md]: thanks, done

File: Concepts.md (hunk 2 above)
Comment (MichalMal99) [reply Concepts.md]: thanks, done

File: Concepts.md (hunk 3 above)
Comment (MichalMal99) [reply Concepts.md]: thanks, done

File: Concepts.md (hunk 5 above)
Comment (MichalMal99) [reply Concepts.md]: thanks, done

File: Concepts.md [hunk 15]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
 We'll be creating our own changelog entry later and will discuss the various elements at that point.  But for now note how the first line contains the package name ('hello'), version number ('2.10-2ubuntu1'), and the Ubuntu release codename ('eoan').
 
 
+### Control file in debian directory
+
+Let's focus on the control file that exists inside the debian
+directory.
+Each control file contains a number of fields. Each line begins with a
+tag such as Package or Version.
```
Comment (cpaelzer) [inline Concepts.md:296]: maybe
tag followed by a Colon such as `Package:` or `Version:`.

File: Concepts.md (hunk 9 above)
Comment (MichalMal99) [reply Concepts.md]: thanks, done

File: Concepts.md (hunk 11 above)
Comment (MichalMal99) [reply Concepts.md]: thanks, done

File: Concepts.md (hunk 6 above)
Comment (MichalMal99) [reply Concepts.md]: thanks, done

File: Concepts.md [hunk 16]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+
+This field typically gives us information on who created the package.
+If we have suggestions or want to propose something, we should talk
+to the maintainer whose email is given in that field.
+Here we see that the maintainer is a person from Debian.
+
+2. Homepage
+
+Here we can find information about the upstream project home page URL.
+We can find more subject matter experts there and some helpful information.
```
Comment (cpaelzer) [inline Concepts.md:334]: "We can" -> "There we might" ?

File: Concepts.md [hunk 17]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+
+3. VCS
+
+There you can find a code and propose a PR, but to do that in this
```
Comment (cpaelzer) [inline Concepts.md:338]: "a code" -> "the code"

File: Concepts.md [hunk 18]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+particular example related to tests, salsa must be present.
```
Comment (cpaelzer) [inline Concepts.md:339]: "related to tests, salsa must be present. In the example, we can easily spot the entry, which tells us that the
 package is developed in git, so there is no salsa."

"related to autopkg tests, the packaging must be part of it (not just the source) and it must allow opening pull requests (PRs) against it. In the example above, we can easily spot the entry, which tells us that the package is developed in git, even with the debian/* content, but there is no way to open and discuss PRs there."

File: Concepts.md [hunk 19]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+There is advice to consider.
+
+For example, we want to propose autopkgtests that are not present in
+Debian, but they are present in Ubuntu and a specific package is developed
+in Git. Should we propose this to upstream, Debian git or the maintainer?
+
+So in that case, we should send it to that specific person from Debian
+(Jörg Frings-Fürst), not submit it to the upstream git repo, because autopkgtests
+exist only in Ubuntu and Debian, and proposing them to the git repository
+is pointless. In that case, no one else apart from Debian
```
Comment (cpaelzer) [inline Concepts.md:353]: pointless -> impossible

File: Concepts.md [hunk 20]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+
+3. VCS
+
+There you can find a code and propose a PR, but to do that in this
+particular example related to tests, salsa must be present.
+In the example, we can easily spot the entry, which tells us that the
+package is developed in git, so there is no salsa.
+
+
+There is advice to consider.
```
Comment (cpaelzer) [inline Concepts.md:344]: "There is advice to consider." -> "Let us play through an example submitting autopkgtest changes:"

File: Concepts.md [hunk 21]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+
+For example, we want to propose autopkgtests that are not present in
+Debian, but they are present in Ubuntu and a specific package is developed
+in Git. Should we propose this to upstream, Debian git or the maintainer?
+
+So in that case, we should send it to that specific person from Debian
+(Jörg Frings-Fürst), not submit it to the upstream git repo, because autopkgtests
```
Comment (cpaelzer) [inline Concepts.md:351]: Maybe break that into bullet points showing the steps of consideration
1. since it is a autopkgtest upstream won't care, so submit it to Debian
2. to get it to Debian we can't submit a PR on the git, so consider mails/bugs
  2.1 Each package is different, some prefer mails, some bugs, some PRs - worst case you have to find out by trying or asking 
3. Depending on the case you might contact the maintainer, file a bug or both

File: Concepts.md [hunk 22]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+is pointless. In that case, no one else apart from Debian
+benefits from it and it's enough when we ease maintenance.
+
+In some other cases, we can send specific PR upstream.
+For example, adding some helpful links to a specific package's page.
+It's quite straightforward.
+
+How do we propose our tests to Debian?
+
+Check if a bug like that exists:
```
Comment (cpaelzer) [inline Concepts.md:362]: "already exists" is what you mean

Move this part up as 3.1 please

**Review by cpaelzer — CHANGES_REQUESTED** [review]
as discussed

File: Concepts.md [hunk 23]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
 We'll be creating our own changelog entry later and will discuss the various elements at that point.  But for now note how the first line contains the package name ('hello'), version number ('2.10-2ubuntu1'), and the Ubuntu release codename ('eoan').
 
 
+### Control file in debian directory
+
+Let's focus on the control file that exists inside the debian
+directory.
+Each control file contains a number of fields. Each line begins with a
```
[the remark this answers is not in the document]
Comment (MichalMal99) [reply Concepts.md:295]: done, thanks

File: Concepts.md (hunk 15 above)
Comment (MichalMal99) [reply Concepts.md:296]: done, thanks

File: Concepts.md (hunk 16 above)
Comment (MichalMal99) [reply Concepts.md:334]: done, thanks

File: Concepts.md (hunk 17 above)
Comment (MichalMal99) [reply Concepts.md:338]: done, thanks

File: Concepts.md (hunk 18 above)
Comment (MichalMal99) [reply Concepts.md:339]: done, thanks

File: Concepts.md (hunk 19 above)
Comment (MichalMal99) [reply Concepts.md:353]: done, thanks

File: Concepts.md (hunk 20 above)
Comment (MichalMal99) [reply Concepts.md:344]: thanks, done

File: Concepts.md [hunk 24]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+How do we propose our tests to Debian?
+
+Check if a bug like that exists:
+https://tracker.debian.org/pkg/ipmitool
+or
+https://bugs.debian.org/cgi-bin/pkgreport.cgi?repeatmerged=no&src=ipmitool
+
+If not, use the second link to open a bug. Follow the steps listed
+there.
+Attach the debdiff to the bug report you created.
```
[the remark this answers is not in the document]
Comment (MichalMal99) [reply Concepts.md:369]: thanks, done

File: Concepts.md [hunk 25]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+
+What to do if there is a salsa?
+
+Add the remote salsa repository and then cherry-pick your commit there, which you 
+should describe and push to a remote repository of salsa.
+
+
```
[the remark this answers is not in the document]
Comment (MichalMal99) [reply Concepts.md:376]: thanks, done

File: Concepts.md [hunk 26]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+(Jörg Frings-Fürst), not submit it to the upstream git repo, because autopkgtests
+exist only in Ubuntu and Debian, and proposing them to the git repository
+is pointless. In that case, no one else apart from Debian
+benefits from it and it's enough when we ease maintenance.
+
+In some other cases, we can send specific PR upstream.
+For example, adding some helpful links to a specific package's page.
+It's quite straightforward.
+
+How do we propose our tests to Debian?
```
[the remark this answers is not in the document]
Comment (MichalMal99) [reply Concepts.md:360]: thanks, done

File: Concepts.md [hunk 27]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+Check if a bug like that exists:
+https://tracker.debian.org/pkg/ipmitool
+or
+https://bugs.debian.org/cgi-bin/pkgreport.cgi?repeatmerged=no&src=ipmitool
+
+If not, use the second link to open a bug. Follow the steps listed
+there.
+Attach the debdiff to the bug report you created.
+
+What to do if there is a salsa?
```
[the remark this answers is not in the document]
Comment (MichalMal99) [reply Concepts.md:371]: thanks, done

File: Concepts.md [hunk 28]
```diff
@@ -288,6 +288,93 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+It's quite straightforward.
+
+How do we propose our tests to Debian?
+
+Check if a bug like that exists:
+https://tracker.debian.org/pkg/ipmitool
+or
+https://bugs.debian.org/cgi-bin/pkgreport.cgi?repeatmerged=no&src=ipmitool
+
+If not, use the second link to open a bug. Follow the steps listed
```
[the remark this answers is not in the document]
Comment (MichalMal99) [reply Concepts.md:367]: thanks, done

File: Concepts.md (hunk 22 above)
Comment (MichalMal99) [reply Concepts.md:362]: thanks, done

File: Concepts.md (hunk 21 above)
Comment (MichalMal99) [reply Concepts.md:351]: thanks, done

File: Concepts.md [hunk 29]
```diff
@@ -288,6 +288,94 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+2. Homepage
+
+There we might find information about the upstream project's home page URL.
+We can find more subject matter experts there and some helpful information.
+
+3. VCS
+
+There you can find a code and maybe propose a pull request for changes.
+In the example, we can easily spot the entry, which tells us that the
+package is developed in git, so there is no salsa.
```
Comment (cpaelzer) [inline Concepts.md:341]: "so there is no salsa"
->
"but not hosted on [salsa](https://salsa.debian.org/)"

File: Concepts.md [hunk 30]
```diff
@@ -288,6 +288,94 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+
+
+
+
+**Let us play through an example of submitting autopkgtest changes:**
+<br>
```
Comment (cpaelzer) [inline Concepts.md:347]: html magic ??

File: Concepts.md [hunk 31]
```diff
@@ -288,6 +288,94 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+(Jörg Frings-Fürst), not submit it to the upstream git repo, because autopkgtests
+exist only in Ubuntu and Debian, and proposing them to the git repository
+is impossible. In that case, no one else apart from Debian
+benefits from it and it's enough when we ease maintenance.
+
+1. Since it is a autopkgtests upstream will not care, so submit it to Debian.
+2. To get it to Debian, we cannot submit a PR on the git, so consider
+mails/creating bug report.<br>
+  2.1. Each package is different, some prefer mails, some bugs, some PRs -worst case you have to find out by trying and asking around.
+3. Depending on the case you might contact the maintainer, file a bug or both.<br>
```
Comment (cpaelzer) [inline Concepts.md:364]: more html magic

File: Concepts.md [hunk 32]
```diff
@@ -288,6 +288,94 @@ Some packages also have a patches/ directory, with .diff or .patch files that wi
...
+  3.1. Check if a bug like that already exists:<br>
+ * https://tracker.debian.org/pkg/ipmitool<br>
+  or<br>
+ * https://bugs.debian.org/cgi-bin/pkgreport.cgi?repeatmerged=no&src=ipmitool<br><br>
+If not, consider creating one using the second link from above.
```
Comment (cpaelzer) [inline Concepts.md:369]: does this need to be indented differntly?

File: Concepts.md (hunk 29 above)
Comment (cpaelzer) [reply Concepts.md]: fixed together

File: Concepts.md (hunk 30 above)
Comment (cpaelzer) [reply Concepts.md]: fixed together

File: Concepts.md (hunk 32 above)
Comment (cpaelzer) [reply Concepts.md]: fixed together

**Comment by cpaelzer** [comment]
In the current form I think this can be helpful to others later on.
Thanks!
No need to re-iterate many more times - merging after the fixup we created together.
